Setup/env_test.sh

```diff
@@ -0,0 +1,136 @@
+SHIELD_NETWORK_ADDR_BLOCK='10.20.0.0/16'
+
+if ! declare -f log_message >/dev/null; then
+    function log_message() {
+        echo "$1"
+        echo "$(date): $1" >>"$LOGFILE"
+    }
+fi
+
+if ! declare -f failed_to_install >/dev/null; then
+    function failed_to_install() {
+        log_message "An error occured during the installation: $1, Exiting!"
+        exit 1
+    }
+fi
+
+function check_url_connectivity() {
+    printf "\nChecking $1 ..."
+    if ! curl "$1" -sS -o /dev/null -w "\nResponse Code: %{http_code}\nDNS time: %{time_namelookup}\nConnection time: %{time_connect}\nPretransfer time: %{time_pretransfer}\nStarttransfer time: %{time_starttransfer}\nTotal time: %{time_total}\n"; then
+        printf "$1 check failed"
+        return 1
+    fi
+}
+
+function check_connectivity() {
+    for url in $URLS_TO_CHECK; do
+        if ! check_url_connectivity "$url"; then
+            echo "Connectivity test failed for $url"
+            return 1
+        fi
+    done
+}
+
+function check_storage_drive_speed() {
+    /sbin/hdparm -Tt $(df -l --output=source /var/lib | awk 'FNR == 2 {print $1}')
+}
+
+function check_free_space() {
+    local FREE_SPACE_ON_ROOT=$(($(stat -f --format="%a*%S" /) / (1024 * 1024 * 1024)))
+    local FREE_SPACE_ON_DEB=$(($(stat -f --format="%a*%S" /var/cache/debconf) / (1024 * 1024 * 1024)))
+    if ((FREE_SPACE_ON_ROOT < MIN_FREE_SPACE_GB)); then
+        failed_to_install "Not enough free space on the / partition. ${FREE_SPACE_ON_ROOT}GB available, ${MIN_FREE_SPACE_GB}GB required."
+    fi
+    if ((FREE_SPACE_ON_DEB < MIN_FREE_SPACE_GB)); then
+        failed_to_install "Not enough free space on the /var/cache/debconf partition. ${FREE_SPACE_ON_DEB}GB available, ${MIN_FREE_SPACE_GB}GB required."
+    fi
+}
+
+function check_network_address_conflicts() {
+    local INTERFACES=($(find /sys/class/net -type l -not -lname '*virtual*' -printf '%f\n'))
+    local INTERFACE_ADDRESSES=()
+
+    for IFACE in "${INTERFACES[@]}"; do
+        INTERFACE_ADDRESSES+=("$(/sbin/ip address show scope global dev $IFACE | grep -oP '(?<=inet )\d+\.\d+\.\d+\.\d+')")
+    done
+
+    for IF_ADDR in "${INTERFACE_ADDRESSES[@]}"; do
+        /usr/bin/python3 <<END
+import ipcalc
+
+if ipcalc.Network("${SHIELD_NETWORK_ADDR_BLOCK}").check_collision(ipcalc.IP("${IF_ADDR}")) :
+    print("WARNING: Address collision detected: ${IF_ADDR} collides with ${SHIELD_NETWORK_ADDR_BLOCK} used by Shield!")
+END
+        check_ip_resolution "${IF_ADDR}"
+    done
+
+}
+
+function check_ip_resolution() {
+    log_message "Trying to resolve $1 to a DNS name..."
+    if ! getent hosts "$1"; then
+        log_message "WARNING: Could not resolve $1!"
+    else
+        log_message "OK"
+    fi
+    return 0
+}
+
+function check_hostname_resolution() {
+    local HOSTNAME="$(hostname)"
+    log_message "Hostname is ${HOSTNAME}"
+    log_message "Trying to resolve ${HOSTNAME} to an IP address..."
+    if ! getent hosts "${HOSTNAME}"; then
```

Comment:
here you are checking that you can resolve your own hostname
but generally this is defined in /etc/hosts 
may be we need to check a hostname that is in the DNS
(or is it already covered by the check_connectivity() function)